### PR TITLE
Move non-core modules to contrib.

### LINKED ISSUE #352: Move modules into contrib
The following modules should be moved into contrib:

* `django_orm`
* `flask_util`
* `keyring_storage`
* `locked_file`
* `multistore_file`
* `appengine`

@nathanielmanistaatgoogle @dhermes I'll be sending a PR shortly for this, but I wanted to make sure you two were comfortable with the list.

### PR DESCRIPTION
**Pull request by theacodes**
Fixes #352

### CODE REVIEW

**Comment by jonparrott** [comment]
Most of this is just moving code around, however, I did have to surgically remove the multstore tests from the file tests.

**Comment by dhermes** [comment]
@jonparrott Please please please please please stop doing "just a little extra" in a PR.

Do the surgery first. Then do this renaming.

**Comment by dhermes** [comment]
Such ocean. Much boil.

**Comment by jonparrott** [comment]
I'm sorry, I know these take lots of time to review. The options were:

* Make `tests.test_file` depend on a `contrib` module, and it being the *only* core test or module that does.
* Move the tests into `tests.contrib.test_multistore_file`, therefore making clean separation between core and contrib.

I really don't like to leave the codebase in a intermediate state such as "moved to contrib, but core still depends on it".

This is the *last* boil the ocean commit. Promise. I am planning to follow through with renaming some of the older storage classes, but I'll do each of those as a separate commit.

**Comment by dhermes** [comment]
Ha, last one! Bold statement.

Why is splitting the tests apart not doable as a pre-amble to this commit?

**Comment by jonparrott** [comment]
> Why is splitting the tests apart not doable as a pre-amble to this commit?

You mean split the tests first, then do this? I mean, I could. If you want.

**Comment by dhermes** [comment]
You should, I do.

This PR should just be a bunch of file renames.

**Comment by jonparrott** [comment]
Will do. Would you prefer two PRs or two commits? 

(This PR will need to change imports and a few mock statements, so it won't be as simple as just file renames)

**Comment by dhermes** [comment]
Two PRs.

**Comment by dhermes** [comment]
Yes I realize it's not as simple as file renames, but the other changes (like ignoring new dirs on `tox.ini`) are simple enough to understand.

**Comment by jonparrott** [comment]
@dhermes this has been rebased and is ready for review.

File: docs/source/oauth2client.rst [hunk 1]
```diff
@@ -13,18 +13,12 @@ Submodules
...
    oauth2client.clientsecrets
    oauth2client.crypt
    oauth2client.devshell
-   oauth2client.django_orm
    oauth2client.file
-   oauth2client.flask_util
    oauth2client.gce
-   oauth2client.keyring_storage
-   oauth2client.locked_file
-   oauth2client.multistore_file
```
Comment (dhermes) [inline docs/source/oauth2client.rst:27]: Are ~~they~~ (the new contrib modules) in the toctree at all? Should I check this out and build the docs?

File: oauth2client/contrib/django_util/__init__.py [hunk 2]
```diff
@@ -292,8 +289,8 @@ def get_authorize_redirect(self):
     def has_credentials(self):
         """Returns True if there are valid credentials for the current user
         and required scopes."""
-        return self.credentials and not self.credentials.invalid \
-               and self.credentials.has_scopes(self.scopes)
+        return (self.credentials and not self.credentials.invalid
+                and self.credentials.has_scopes(self.scopes))
```
Comment (dhermes) [inline oauth2client/contrib/django_util/__init__.py:293]: :+1:

File: tests/contrib/test_appengine.py [hunk 3]
```diff
@@ -34,7 +34,7 @@
 import mock
 import webapp2
 
-from .http_mock import CacheMock
+from ..http_mock import CacheMock
```
Comment (dhermes) [inline tests/contrib/test_appengine.py:37]: Ew. Already was icky. (Nothing for you to do, just my sentiment.)

File: tests/contrib/test_keyring_storage.py [hunk 4]
```diff
@@ -68,7 +65,8 @@ def test_locked_get(self):
                                return_value=mock_content,
                                autospec=True) as get_password:
             with mock.patch(
-                    'oauth2client.keyring_storage.Credentials') as MockCreds:
+                'oauth2client.contrib.keyring_storage.Credentials') \
```
Comment (dhermes) [inline tests/contrib/test_keyring_storage.py]: Just say no to `\`.

``` python
class_name = 'oauth2client.contrib.keyring_storage.Credentials'
with mock.patch(class_name) as MockCreds:
    ...
```

---

Also, even you can prevent forest fires.

File: tox.ini [hunk 5]
```diff
@@ -27,18 +27,18 @@ commands =
       --cover-erase \
       --cover-tests \
       --cover-branches \
-      --ignore-files=test_appengine\.py
+      --ignore-files=contrib/test_appengine\.py
```
Comment (dhermes) [inline tox.ini:30]: AFAIK `nose` only knows about filenames, not paths.

File: tox.ini [hunk 6]
```diff
@@ -27,18 +27,18 @@ commands =
...
     nosetests \
       --with-coverage \
-      --cover-package=oauth2client.appengine \
-      --cover-package=tests.test_appengine \
+      --cover-package=oauth2client.contrib.appengine \
+      --cover-package=tests.contrib.test_appengine \
```
Comment (dhermes) [inline tox.ini:34]: Except `nosecover` breaks the mold of the thing I said above.

**Comment by dhermes** [comment]
Finished review. Looks mostly good. Ping me once done with changes (I asked for at least one, but maybe only one, if that's the case I apologize for making **changes** plural).

File: tests/contrib/test_keyring_storage.py (hunk 4 above)
Comment (jonparrott) [inline tests/contrib/test_keyring_storage.py]: There's no other way to wrap this. You can't do `with (expr as value)`, it's inexplicably invalid syntax.

Comment (theacodes) [inline tests/contrib/test_keyring_storage.py]: Done.

File: tox.ini (hunk 5 above)
Comment (theacodes) [inline tox.ini:30]: Seems fine either way, better to be explicit?

Comment (dhermes) [inline tox.ini:30]: Sure. I seem to remember trying to use a full path before and it not working. But it's not clear what I'm actually remembering.

**Comment by dhermes** [comment]
LGTM

**Comment by dhermes** [comment]
@jonparrott Did you mean to erase the `appengine` module in this PR?

**Comment by dhermes** [comment]
NVM I am stupid

**Comment by dhermes** [comment]
```python
======================================================================
ERROR: Failure: ImportError (No module named dev_appserver)   
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dhermes/google_contracting/oauth2client/.tox/cover/local/lib/python2.7/site-packages/nose/loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "/home/dhermes/google_contracting/oauth2client/.tox/cover/local/lib/python2.7/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/home/dhermes/google_contracting/oauth2client/.tox/cover/local/lib/python2.7/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/dhermes/google_contracting/oauth2client/tests/contrib/test_appengine.py", line 32, in <module>
    import dev_appserver
ImportError: No module named dev_appserver
-------------------- >> begin captured logging << --------------------
py.warnings: WARNING: /home/dhermes/google_contracting/oauth2client/.tox/cover/local/lib/python2.7/site-packages/django/db/models/fields/subclassing.py:22: RemovedInDjango110Warning: SubfieldBase has been 
deprecated. Use Field.from_db_value instead.
  RemovedInDjango110Warning)

```

**Comment by dhermes** [comment]
OK maybe I'm not crazy since 

```
nosetests --with-coverage --cover-package=oauth2client --cover-package=tests \
--cover-erase --cover-tests --cover-branches --ignore-files=contrib/test_appengine\.py
```

should not result in an import error in 

```
.../oauth2client/tests/contrib/test_appengine.py
```

**Comment by jonparrott** [comment]
Ugh, app engine. Created a new PR #363.

**Comment by dhermes** [comment]
Thanks. AFAICT the issue was actually the slash that we discussed.

**Pull request merged**